From 32d62d34fe52ab38ee571c34ee0590c223d44bca Mon Sep 17 00:00:00 2001
From: Thomas Stuart <tjstuart@google.com>
Date: Thu, 12 Jun 2025 17:50:08 +0000
Subject: [PATCH] DO NOT MERGE remove the contacts picker from
 CallForwardingOptions

For older versions of Android, UpsideDownCake (14) and below,
the contacts picker should be removed since there is no good
API for verifying the calling process.

Flag: EXEMPT security fix
Bug: 337781186
Test: manual
Change-Id: I0f49ffa31dcabc07281ea4e10471cd3f3d1364ca
---
 .../phone/CallForwardEditPreference.java      |  1 -
 .../phone/GsmUmtsCallForwardOptions.java      | 41 +------------------
 2 files changed, 2 insertions(+), 40 deletions(-)

diff --git a/src/com/android/phone/CallForwardEditPreference.java b/src/com/android/phone/CallForwardEditPreference.java
index d1c8303be..a125f7717 100644
--- a/src/com/android/phone/CallForwardEditPreference.java
+++ b/src/com/android/phone/CallForwardEditPreference.java
@@ -73,7 +73,6 @@ public class CallForwardEditPreference extends EditPhoneNumberPreference {
         reason = a.getInt(R.styleable.CallForwardEditPreference_reason,
                 CommandsInterface.CF_REASON_UNCONDITIONAL);
         a.recycle();
-
         Log.d(LOG_TAG, "mServiceClass=" + mServiceClass + ", reason=" + reason);
     }
 
diff --git a/src/com/android/phone/GsmUmtsCallForwardOptions.java b/src/com/android/phone/GsmUmtsCallForwardOptions.java
index 6a5cb83d5..97e7264ad 100644
--- a/src/com/android/phone/GsmUmtsCallForwardOptions.java
+++ b/src/com/android/phone/GsmUmtsCallForwardOptions.java
@@ -208,45 +208,8 @@ public class GsmUmtsCallForwardOptions extends TimeConsumingPreferenceActivity {
             Log.d(LOG_TAG, "onActivityResult: contact picker result not OK.");
             return;
         }
-        Cursor cursor = null;
-        try {
-            // check if the URI returned by the user belongs to the user
-            final int currentUser = UserHandle.getUserId(Process.myUid());
-            if (currentUser
-                    != ContentProvider.getUserIdFromUri(data.getData(), currentUser)) {
-
-                Log.w(LOG_TAG, "onActivityResult: Contact data of different user, "
-                        + "cannot access");
-                return;
-            }
-            cursor = getContentResolver().query(data.getData(),
-                NUM_PROJECTION, null, null, null);
-            if ((cursor == null) || (!cursor.moveToFirst())) {
-                Log.d(LOG_TAG, "onActivityResult: bad contact data, no results found.");
-                return;
-            }
-
-            switch (requestCode) {
-                case CommandsInterface.CF_REASON_UNCONDITIONAL:
-                    mButtonCFU.onPickActivityResult(cursor.getString(0));
-                    break;
-                case CommandsInterface.CF_REASON_BUSY:
-                    mButtonCFB.onPickActivityResult(cursor.getString(0));
-                    break;
-                case CommandsInterface.CF_REASON_NO_REPLY:
-                    mButtonCFNRy.onPickActivityResult(cursor.getString(0));
-                    break;
-                case CommandsInterface.CF_REASON_NOT_REACHABLE:
-                    mButtonCFNRc.onPickActivityResult(cursor.getString(0));
-                    break;
-                default:
-                    // TODO: may need exception here.
-            }
-        } finally {
-            if (cursor != null) {
-                cursor.close();
-            }
-        }
+        Log.w(LOG_TAG, "onActivityResult: skipping contact picker result since this"
+                + " functionality is no longer supported.");
     }
 
     @Override
-- 
2.34.1

